lib/elixir/lib/string_io.ex

```diff
@@ -359,6 +344,28 @@ defmodule StringIO do
 
   ## helpers
 
+  defp state_after_read(%{capture_prompt: false} = s, remainder, _prompt) do
+    %{s | input: remainder}
+  end
+
+  defp state_after_read(%{capture_prompt: true, output: output} = s, remainder, prompt) do
+    %{s | input: remainder, output: <<output::binary, IO.chardata_to_string(prompt)::binary>>}
+  end
+
+  defp bytes_until_eol("", _, count), do: {:ok, count}
+  defp bytes_until_eol(<<"\r\n"::binary, _::binary>>, _, count), do: {:ok, count + 2}
```

Comment:
We get the `\r` information here so could pass this information in the return tuple?